src/gui/vector/qgsattributesformproperties.cpp

```diff
@@ -1988,3 +2010,283 @@ void QgsAttributesFormProperties::updatedFields()
+
+  // Widget general settings
+  QDomElement widgetGeneralSettingsElem = doc.createElement( QStringLiteral( "widgetGeneralSettings" ) );
+  widgetGeneralSettingsElem.setAttribute( QStringLiteral( "editable" ), !mLayer->editFormConfig().readOnly( index ) );
+  widgetGeneralSettingsElem.setAttribute( QStringLiteral( "reuse_last_values" ), mLayer->editFormConfig().labelOnTop( index ) );
+  widgetGeneralSettingsElem.setAttribute( QStringLiteral( "label_on_top" ), mLayer->editFormConfig().reuseLastValue( index ) );
+  documentElement.appendChild( widgetGeneralSettingsElem );
+
+  // Widget display section
+  if ( mAttributeWidgetEdit )
+  {
+    // Go for the corresponding form layout item and extract its display settings
+    if ( mFormLayoutTree->selectedItems().count() != 1 )
+      return;
+
+    const QTreeWidgetItem *itemLayout = mFormLayoutTree->selectedItems().at( 0 );
+    const DnDTreeItemData itemDataLayout = itemLayout->data( 0, DnDTreeRole ).value<DnDTreeItemData>();
+
+    QDomElement displayElement = doc.createElement( QStringLiteral( "widgetDisplay" ) );
+    displayElement.setAttribute( QStringLiteral( "showLabel" ), itemDataLayout.showLabel() );
+    displayElement.setAttribute( QStringLiteral( "horizontalStretch" ), itemDataLayout.horizontalStretch() );
+    displayElement.setAttribute( QStringLiteral( "verticalStretch" ), itemDataLayout.verticalStretch() );
+    displayElement.appendChild( itemDataLayout.labelStyle().writeXml( doc ) );
+    documentElement.appendChild( displayElement );
+  }
+
+  doc.appendChild( documentElement );
+
+  QMimeData *mimeData = new QMimeData;
+  mimeData->setData( QStringLiteral( "application/x-qgsattributetabledesignerelementclipboard" ), doc.toByteArray() );
+  QClipboard *clipboard = QApplication::clipboard();
+  clipboard->setMimeData( mimeData );
+}
+
+void QgsAttributesFormProperties::pasteWidgetConfiguration()
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+
+  if ( mAvailableWidgetsTree->selectedItems().count() != 1 )
+    return;
+
+  QTreeWidgetItem *item = mAvailableWidgetsTree->selectedItems().at( 0 );
+
+  const QString fieldName = item->data( 0, FieldNameRole ).toString();
+  const int fieldIndex = mLayer->fields().indexOf( fieldName );
+
+  if ( fieldIndex < 0 )
+    return;
+
+  // Get base config from target item and ovewrite settings when possible
+  FieldConfig config = item->data( 0, FieldConfigRole ).value<FieldConfig>();
+
+  QDomDocument doc;
+  QClipboard *clipboard = QApplication::clipboard();
+  if ( doc.setContent( clipboard->mimeData()->data( QStringLiteral( "application/x-qgsattributetabledesignerelementclipboard" ) ) ) )
+  {
+    QDomElement docElem = doc.documentElement();
+    if ( docElem.tagName() != QLatin1String( "FormWidgetClipboard" ) )
+      return;
+
+    // We avoid pasting if origin and target fields are the same
+    const QString sourceFieldName = docElem.attribute( QStringLiteral( "name" ) );
+    if ( sourceFieldName == fieldName )
```

Comment:
could this be checked when populating the menu?

Reply:
Yes, it should be there.

However, perhaps we can discuss it with A. Poncet because it also might make sense to copy a field's config to keep it in the clipboard, do some changes, visualize and test changes, and after that, return to what we have in the clipboard (i.e., paste the config in the same field as a way to undo things). I'll post this as a comment in the original issue.

Reply:
Posted as a comment in the original issue to continue the discussion.